### PR TITLE
Add OIDC Bearer Token Auth mechanism based on the PoC design

### LINKED ISSUE #206: Add OIDC Bearer Token Auth mechanism based on the PoC design
**Overview of the Feature Request**

The PoC helped us validate the OIDC auth design. The goal now is to carry out the refined and actual implementation. Specifically, in this issue, the goal is to implement the changes related to js-dataverse.

Link to the PoC: https://github.com/IQSS/dataverse-frontend/pull/516

**What kind of user is the feature intended for?**
SPA users

**What inspired the request?**

- https://github.com/IQSS/dataverse-frontend/pull/516
- Q4 milestone

**What existing behavior do you want changed?**

None

**Any brand-new behavior do you want to add to Dataverse?**

OIDC auth

**Any open or closed issues related to this feature request?**

 https://github.com/IQSS/dataverse-frontend/pull/516

### PR DESCRIPTION
**Pull request by g-saracca**
## What this PR does / why we need it:
Adds OIDC Bearer Token Auth mechanism based on the PoC design.
Also adds a `RegisterUser` use case based on the new api endpoint added in https://github.com/IQSS/dataverse/pull/10972 

## Which issue(s) this PR closes:

- Closes #206 

## Suggestions on how to test this:
This issue just adds a new mechanism to js-dataverse use cases, there is a unit test to check that the bearer token is being sent correctly in the Authorization header when configured with a bearer token.
Also a unit test for the RegisterUser use case has been added, more in-depth testing will be done on the SPA integration.

### CODE REVIEW

File: src/core/infra/repositories/apiConfigBuilders.ts [hunk 1]
```diff
@@ -13,11 +13,14 @@ export const buildRequestConfig = (
     headers: {
       'Content-Type': contentType
     },
+    withCredentials: false,
```
Comment (ekraffmiller) [inline src/core/infra/repositories/apiConfigBuilders.ts]: In` AxiosRequestConfig`, the default value of `withCredentials` is `false`, so would it be cleaner to not set this here, and only set the value to true for the SESSION_COOKIE use case?  It works fine as is, it was just a little confusing to me at first.

**Review by ekraffmiller — COMMENTED** [review]
Looks good!  I just have a question about setting `withCredentials`

**Review by ekraffmiller — APPROVED** [review]
Looks good!

**Comment by ofahimIQSS** [comment]
Tests are passing - merging PR

**Pull request merged**